### PR TITLE
feat: an example showing how to use Cloud Run

### PR DESCRIPTION
**Pull request by coryan**
A simple "Hello World" example for C++ on Cloud Run.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/3352)

### CODE REVIEW

File: google/cloud/examples/cloud_run_hello/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Pick the project to run the Cloud Run Deployment.
+
+```bash
+export GOOGLE_CLOUD_PROJECT=...
+```
+
+This script will enable the necessary APIs, build the docker image used by
+Cloud Run using Cloud Build, create a service account for the Cloud Run
+deployment,
+Run the program to bootstrap the indexer, this will create the Cloud Spanner
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/examples/cloud_run_hello/README.md]: Nope, removed.

File: google/cloud/examples/cloud_run_hello/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ~~~
+
+cmake_minimum_required(VERSION 3.10)
+
+# Define the project name and where to report bugs.
+set(PACKAGE_BUGREPORT "https://github.com/coryan/docker-grpc-cpp/issues")
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/examples/cloud_run_hello/CMakeLists.txt]: Fixed.

File: google/cloud/examples/cloud_run_hello/cloud_run_hello.cc [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+[[noreturn]] void throw_error(be::error_code ec, char const* what) {
+  std::ostringstream os;
+  os << what << ": " << ec.message() << "\n";
+  throw std::runtime_error(os.str());
+}
+
+/**
+ * Handle a HTTP request.
+ */
+class http_handler {
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/examples/cloud_run_hello/cloud_run_hello.cc]: Fixed, here and elsewhere.

**Comment by coryan** [comment]
@devjgm PTAL.

**Comment by devjgm** [comment]
I mentioned:

> There's something about the `google/cloud/examples` directory that doesn't feel quite right to me (maybe it's me). What do you think about moving this to a top-level `examples/` directory.

Any thoughts on moving the location of this examples dir?

**Comment by coryan** [comment]
Moving to `examples/` sounds like a good idea, I will leave this to a future PR (see #3357) because it would affect ci scripts, and the existing gcs2cbt example, and maybe more.

**Comment by devjgm** [comment]
SGTM.

**Pull request merged**